cloud/scope/cluster.go

```diff
@@ -0,0 +1,147 @@
+	if params.Cluster == nil {
+		return nil, errors.New("Cluster is required when creating a ClusterScope")
+	}
+	if params.DOCluster == nil {
+		return nil, errors.New("DOCluster is required when creating a ClusterScope")
+	}
+	if params.Logger == nil {
+		params.Logger = klogr.New()
+	}
+
+	session, err := params.DOClients.Session()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create DO session")
+	}
+
+	if params.DOClients.Actions == nil {
+		params.DOClients.Actions = session.Actions
+	}
+
+	if params.DOClients.Droplets == nil {
+		params.DOClients.Droplets = session.Droplets
+	}
+
+	if params.DOClients.Images == nil {
+		params.DOClients.Images = session.Images
+	}
+
+	if params.DOClients.Keys == nil {
+		params.DOClients.Keys = session.Keys
+	}
+
+	if params.DOClients.LoadBalancers == nil {
+		params.DOClients.LoadBalancers = session.LoadBalancers
+	}
+
+	helper, err := patch.NewHelper(params.DOCluster, params.Client)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to init patch helper")
+	}
+
+	return &ClusterScope{
+		Logger:      params.Logger,
+		client:      params.Client,
+		DOClients:   params.DOClients,
+		Cluster:     params.Cluster,
+		DOCluster:   params.DOCluster,
+		patchHelper: helper,
+	}, nil
+}
+
+// ClusterScope defines the basic context for an actuator to operate upon.
+type ClusterScope struct {
+	logr.Logger
+	client      client.Client
+	patchHelper *patch.Helper
+
+	DOClients
+	Cluster   *clusterv1.Cluster
+	DOCluster *infrav1.DOCluster
+}
+
+// Close closes the current scope persisting the cluster configuration and status.
+func (s *ClusterScope) Close() error {
+	return s.patchHelper.Patch(context.TODO(), s.DOCluster)
+}
+
+// Name returns the cluster name.
+func (s *ClusterScope) Name() string {
+	return s.Cluster.GetName()
+}
+
+// Namespace returns the cluster namespace.
+func (s *ClusterScope) Namespace() string {
+	return s.Cluster.GetNamespace()
+}
+
+// Region returns the cluster region.
+func (s *ClusterScope) Region() string {
+	return s.DOCluster.Spec.Region
+}
+
+// Network returns the cluster network object.
+func (s *ClusterScope) Network() *infrav1.DONetworkResource {
+	return &s.DOCluster.Status.Network
+}
+
+// SetReady sets the DOCluster Ready Status
+func (s *ClusterScope) SetReady() {
+	s.DOCluster.Status.Ready = true
+}
+
+// SetAPIEndpoint sets the DOCluster status APIEndpoints.
+func (s *ClusterScope) SetAPIEndpoint(apiEndpoint []infrav1.APIEndpoint) {
+	s.DOCluster.Status.APIEndpoints = apiEndpoint
+}
+
+// APIServerLoadbalaners get the DOCluster Spec Network APIServerLoadbalaners.
+func (s *ClusterScope) APIServerLoadbalaners() *infrav1.DOLoadbalancers {
+	return &s.DOCluster.Spec.Network.APIServerLoadbalancers
+}
+
+// APIServerLoadbalanersRef get the DOCluster status Network APIServerLoadbalancersRef.
+func (s *ClusterScope) APIServerLoadbalanersRef() *infrav1.DOResourceReference {
+	return &s.DOCluster.Status.Network.APIServerLoadbalancersRef
+}
```

Comment:
The function names have a typo. It should be `Loadbalancers` instead of `Loadbalaners`.

Reply:
addressed, thanks